avr-hal-generic/src/port.rs

```diff
@@ -137,6 +143,28 @@ impl<PIN: PinOps, MODE: mode::Io> Pin<MODE, PIN> {
         }
     }
 
+    /// Convert this pin into an open-drain output pin, setting the state to low.
+    /// See [Digital Output Open Drain](#digital-output-open-drain)
+    pub fn into_opendrain(mut self) -> Pin<mode::OpenDrain, PIN> {
+        unsafe { self.pin.out_clear() };
+        unsafe { self.pin.make_output() };
+        Pin {
+            pin: self.pin,
+            _mode: PhantomData,
+        }
+    }
+
+    /// Convert this pin into an open-drain output pin, setting the state to high.
+    /// See [Digital Output Open Drain](#digital-output-open-drain)
+    pub fn into_opendrain_high(mut self) -> Pin<mode::OpenDrain, PIN> {
+        unsafe { self.pin.out_clear() };
+        unsafe { self.pin.make_input(false) };
```

Comment:
The `out_clear()` here is not needed, `make_input()` already takes care of it:

https://github.com/Rahix/avr-hal/blob/5d757696e338104622a0f89c51020b718bf48b62/avr-hal-generic/src/port.rs#L563-L572

Doing it here is actually harmful because it will lead to a glitch on the electrical pin:  If the pin was previously a HIGH output, it will briefly go low before entering the floating state. `make_input()` has the order correct such that this glitch cannot occur.